### PR TITLE
Set frozen/cold tier for Security ML integrations

### PR DESCRIPTION
**Pull request by susan-shu-c**
## Proposed commit message

Per discussion brought up by @SourinPaul , here is a draft PR illustrating what the packages would look like with the data frozen tier and data cold tier filtered out. The PR also lints/formats these files with `elastic-package` to make sure the additional field conforms to style guides.

However, we must first discuss with the authors of those respective packages if that is the **expected behavior** before merging. @ajosh0504 @sodhikirti07 [Edit] We've discussed these changes and gotten approval.

Affected packages:
- [x] ProblemChild
- [x] DGA
- [x] LMD
- [x] DED

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

Additional info:
> Apoorva - Agree with the suggestion [...] to exclude cold/frozen tier from the module query. We seem to be doing it in our OOTB jobs in Kibana as well, so looks like the way to go to prevent ML jobs from hitting those tiers. Eg: https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/server/models/data_recognizer/modules/security_network/manifest.json#L17

## Related issues

-

### CODE REVIEW

**Comment by ajosh0504** [comment]
Thanks for getting this started! Changes look good, but you'd have to update the package versions as well, specifically, you'd add a new entry to the changelog and bump the version in the manifest file. See [this](https://github.com/elastic/integrations/pull/8473) PR for reference.

**Review by ajosh0504 — CHANGES_REQUESTED** [review]
Thanks for getting this started! Changes look good, but you'd have to update the package versions as well, specifically, you'd add a new entry to the changelog and bump the version in the manifest file. See [this](https://github.com/elastic/integrations/pull/8473) PR for reference.

**Comment by susan-shu-c** [comment]
Re: @ajosh0504 
Ah oops, forgot that! I will do it if the changes look good to Kirti for her packages too

File: packages/lmd/manifest.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 format_version: 3.0.0
 name: lmd
 title: "Lateral Movement Detection"
-version: 2.1.1
+version: 2.1.2
```
Comment (sodhikirti07) [inline packages/lmd/manifest.yml:4]: You'll need to add below under `transform.yml` for LMD:
  `fleet_transform_version: 2.1.2`

Comment (susan-shu-c) [reply packages/lmd/manifest.yml:4]: Done!

**Review by droberts195 — APPROVED** [review]
LGTM

Thanks for adding these extra conditions.

**Review by sodhikirti07 — APPROVED** [review]
Changes lgtm!

**Comment by susan-shu-c** [comment]
Buildkite's showing this error for DGA:

```

023/11/15 20:00:39 DEBUG Build zipped package
--
  | 2023/11/15 20:00:39 DEBUG Compress using archiver.Zip (destination: /opt/buildkite-agent/builds/bk-agent-prod-gcp-1700078074357728088/elastic/integrations/build/packages/dga-2.0.1.zip)
  | 2023/11/15 20:00:39 DEBUG Create work directory for archiving: /tmp/elastic-package-1799526892/dga-2.0.1
  | 2023/11/15 20:00:39 DEBUG Validating built .zip package (path: /opt/buildkite-agent/builds/bk-agent-prod-gcp-1700078074357728088/elastic/integrations/build/packages/dga-2.0.1.zip)
  | 2023/11/15 20:00:39 DEBUG POST https://127.0.0.1:5601/api/fleet/epm/packages
  | Error: can't install the package: could not zip-install package; API status code = 429; response body = {"statusCode":429,"error":"Too Many Requests","message":"[parent] Data too large, data for [<http_request>] would be [1029186428/981.5mb], which is larger than the limit of [1020054732/972.7mb], real usage: [935794648/892.4mb], new bytes reserved: [93391780/89mb], usages [inflight_requests=93391780/89mb, request=0/0b, fielddata=288/288b, eql_sequence=0/0b, model_inference=246784104/235.3mb]: circuit_breaking_exception\n\tRoot causes:\n\t\tcircuit_breaking_exception: [parent] Data too large, data for [<http_request>] would be [1029186428/981.5mb], which is larger than the limit of [1020054732/972.7mb], real usage: [935794648/892.4mb], new bytes reserved: [93391780/89mb], usages [inflight_requests=93391780/89mb, request=0/0b, fielddata=288/288b, eql_sequence=0/0b, model_inference=246784104/235.3mb]"}
  | [dga] run_tests_package failed
  | Collect Elastic stack logs
  | 2023/11/15 20:00:46  WARN CommitHash is undefined, in both /opt/buildkite-agent/.elastic-package/version and the compiled binary, config may be out of date.
```

When I spin up a local stack to test, I can install DGA however:

Screenshot 2023-11-15 at 2 58 51 PM

Screenshot 2023-11-15 at 2 52 30 PM

The other packages are fine as well on Buildkite. Might test again later to see if the tests can succeed.

**Comment by droberts195** [comment]
> Buildkite's showing this error for DGA

It looks like BuildKite is testing with a small ES node (1GB JVM heap) and there's too much going on simultaneously for that size of node. I don't think the changes in this PR caused this.

**Comment by susan-shu-c** [comment]
Sounds good, won't merge today (Nov 14) end of day, but aiming for tomorrow morning (Nov 15) to try again and merge. But otherwise this should be good.

**Pull request merged**